src/pybind/mgr/dashboard/frontend/src/app/shared/directives/stateful-tab.directive.ts

```diff
@@ -8,14 +8,17 @@ import { NgbNav, NgbNavChangeEvent } from '@ng-bootstrap/ng-bootstrap';
 export class StatefulTabDirective implements OnInit {
   @Input()
   cdStatefulTab: string;
+  @Input()
+  cdStatefulTabDefault = '';
 
   private localStorage = window.localStorage;
 
   constructor(@Optional() @Host() private nav: NgbNav) {}
 
   ngOnInit() {
     // Is an activate tab identifier stored in the local storage?
-    const activeId = this.localStorage.getItem(`tabset_${this.cdStatefulTab}`);
+    const activeId =
+      this.cdStatefulTabDefault || this.localStorage.getItem(`tabset_${this.cdStatefulTab}`);
```

Comment:
Argghhh... we are keeping state by storing stuff in the `localStorage`? This is a very nasty hack (with downsides, like causing issues with multi-tab browsing), probably inherited from the original Dashboard.

Angular (and most web frameworks) provide mechanisms for pages to set/get state, and the browser itself manages this state through the browser history:
* Exposing the state in the URL (e.g.: if we click on a tab, a `#tabName` is added to the URL). That way, whenever we browse back, we return to the exact same view.
  * This was a cumbersome task, as it's not trivial to instantiate an Angular component from the `Router` params at `onInit` stage. But, since [Angular 16](https://netbasal.com/binding-router-information-to-routed-component-inputs-in-angular-78ee92f63e64), this has dramatically improved, and now there's an option that allows to pass URL query params as the inputs for components.
* Using `routerLink` [`state`](https://angular.io/api/router/NavigationBehaviorOptions#state), to save custom data, which Angular automatically store in the [browser history API](https://developer.mozilla.org/en-US/docs/Web/API/History_API).

There are other [trickier approaches](https://itnext.io/cache-components-with-angular-routereusestrategy-3e4c8b174d5f),

Reply:
yeah.. this would need a quite a bit of refactoring. I just modified little bit to make it work/fix for my usecase. I'll create tracker and see if we can change the behvaiour

Reply:
https://tracker.ceph.com/issues/62267